share/src/main/java/org/apache/kafka/server/share/DeleteShareGroupStateParameters.java

```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.kafka.server.group.share;
+package org.apache.kafka.server.share;
```

Comment:
Should we structure the classes in `group` package inside `share` module itself?

Reply:
These have little to do with share group and more with share coordinator.
Additionally we will not have any other groups in share module so would not add much.

Reply:
Then should it be in `coordinator` package in `share`? I am just trying to get it better structured rather pushing everything in parent package.

Reply:
For the callers the main entry point is NoOpStatePersister/Persister/DefaultStatePersister. 
All the other classes are wrappers for the actual RPC records. 

Ideally callers should not we aware that there is even a coordinator. They will however need to create objects to pass parameters into and fetch results from these interfaces.

The thinking behind the current structure was not to expose to the callers that there is a specific handler/share coordinator which these interfaces delegate the call to.

We followed similar approach in coordinator-common where we just kept one package for Records/metrics/runtime etc.

Reply:
Ok, not fully convinced though but we can see if required specific sub package later.